### PR TITLE
P4-OVS: Update grpc and protobuf modules to the latest version

### PR DESCRIPTION
**Pull request by nupuruttarwar**
This patch does the following:

Update grpc and protobuf modules to the latest version, thereby avoiding
security vulnerability associated with previous grpc version used 1.17.2
Scripts cleanup

Title: Update grpc and protobuf modules to the latest version
Change-type: DefectResolution
Signed-off-by: nupuruttarwar nupur.uttarwar@intel.com

### CODE REVIEW

**Comment by mestery** [comment]
Similar to what @jafingerhut and I noted on https://github.com/ipdk-io/ipdk/pull/84, it would be great to get these patches merged into the upstream projects rather than carrying local patches.

**Comment by jafingerhut** [comment]
Isn't this PR removing a patch file?  i.e. moving in the desired direction?

**Comment by mestery** [comment]
D'oh, it is! Good catch @jafingerhut, and nice work @nupuruttarwar!

**Review by mestery — APPROVED** [review]
Similar to wha

File: fix_sde_libs.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ then
     exit 1
 fi
 SDE_INSTALL_PATH=$1
-la_files=(./install/lib/*.la)
+la_files=($1/install/lib/*.la)
```
Comment (rst0git) [inline fix_sde_libs.sh]: `"$SDE_INSTALL_PATH/install/lib/*.la"`

Comment (nupuruttarwar) [reply fix_sde_libs.sh]: yes that's the expected location for la files, earlier it was incorrect

File: install_dep_packages.sh [hunk 2]
```diff
@@ -134,10 +130,11 @@ echo "####  Cloning, Building and Installing the '$MODULE' module ####"
 mkdir -p "${SRC_DIR}"/"$MODULE"
 git clone https://github.com/google/protobuf.git "$SRC_DIR"/"$MODULE"
 cd "$SRC_DIR"/"$MODULE"
-git checkout tags/v3.6.1
-./autogen.sh
-./configure "$CONFIG_PREFIX"
-make "$NUM_THREADS"
+git checkout tags/v3.18.1
```
Comment (rst0git) [inline install_dep_packages.sh]: The following command would reduce the amount of data (and time) it takes to download the `v3.18.1` branch.
```bash
git clone --depth=1 -b v3.18.1 https://github.com/google/protobuf.git "$SRC_DIR"/"$MODULE"
```

File: install_dep_packages.sh [hunk 3]
```diff
@@ -147,14 +144,17 @@ echo "####  Cloning, Building and Installing the '$MODULE' module ####"
 mkdir -p "${SRC_DIR}"/"$MODULE"
 git clone https://github.com/google/grpc.git "$SRC_DIR"/"$MODULE"
```
Comment (rst0git) [inline install_dep_packages.sh]:
```bash
git clone --depth=1 -b v1.42.0 https://github.com/google/grpc.git "$SRC_DIR"/"$MODULE"
```

File: install_dep_packages.sh (hunk 2 above)
Comment (nupuruttarwar) [reply install_dep_packages.sh]: @rst0git I wasn't aware of depth parameter. Good to know, thanks for sharing!! Will try to use it wherever possible!

File: fix_sde_libs.sh (hunk 1 above)
Comment (stolsma) [inline fix_sde_libs.sh]: Why not use (the line above defined) `$SDE_INSTALL_PATH` instead of `$1` ? Makes the code clearer.

**Review by stolsma — CHANGES_REQUESTED** [review]
Can't we do the `git clone --depth=1 -b ` trick for the other dependencies too? 🤔

**Comment by nupuruttarwar** [comment]
> Can't we do the `git clone --depth=1 -b ` trick for the other dependencies too? 🤔
I tried that but depth only allows branch name to be specified and for other dependencies we are checking out a particular tag. Looking into it, if there is another trick which can be used here.

**Comment by stolsma** [comment]
> > I tried that but depth only allows branch name to be specified and for other dependencies we are checking out a particular tag. Looking into it, if there is another trick which can be used here.

It looks like git doesn't support using hash tags for -b... 😔

**Comment by nupuruttarwar** [comment]
> > > I tried that but depth only allows branch name to be specified and for other dependencies we are checking out a particular tag. Looking into it, if there is another trick which can be used here.
> 
> It looks like git doesn't support using hash tags for -b... 😔
>>> Yes, maybe we can find a nearest branch that contains this commit and specify that when cloning using depth. Not sure if there are other caveats of doing this, thoughts?

**Review by namratalimaye — APPROVED** [review]

**Pull request merged**